src/ModuleLoaderInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Dhii\Modular;
+
+use Traversable;
+
+/**
+ * Something that can load modules.
+ *
+ * Loading should ideally be mostly determined by the module type.
+ *
+ * Since a module is a generic concept, its use can vary from as simple as representing a file to be loaded to a
+ * complete sub-system that requires a substantial amount of initialisation.
+ *
+ * As such, implementors are encouraged to respect the different module types that can exist and load them accordingly.
+ *
+ * @since [*next-version*]
+ */
+interface ModuleLoaderInterface
+{
+    /**
+     * Loads the given module or collection of modules.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|Traversable|ModuleInterface $modules A module or a collection of modules. Default: array()
+     */
+    public function load($modules = array());
```

Comment:
It's already "unioned" in the docs, no? And yes, I agree that determining load order sounds like something that should be the responsibility of the loader. This way, a `ModuleInterface` descendant can expose a list of "dependencies" and the loader can load them based on that. Or maybe it can specify a load order priority value, and again the load order takes care of that. But I would prefer it narrowed down to except _exactly_ `ModuleInterface[]|Traversable`. In the sense that even the normalization of a single module is something that makes the handling... not so clean, so to speak.